node/internal_binding/handle_wrap.ts

```diff
@@ -34,7 +34,7 @@ export class HandleWrap extends AsyncWrap {
 
   close(cb: () => void = () => {}) {
     this._onClose();
-    cb();
+    queueMicrotask(cb);
```

Comment:
I also tried some other patterns:
- `nextTick(cb)` didn't work (the events happen in `close` -> `error` order)
- `setTimeout(cb, 0)` worked